lib/chef/resource/aws_subnet.rb

```diff
@@ -75,6 +76,12 @@ class Chef::Resource::AwsSubnet < Chef::Provisioning::AWSDriver::AWSResourceWith
   #
   attribute :route_table#, kind_of: [ String, AwsRouteTable, AWS::EC2::RouteTable ], equal_to: [ :default_to_main ]
 
+  #
+  # The Network ACL to associate with this subnet. Subnets may only
+  # be associated with a single Network ACL.
+  #
+  attribute :network_acl, kind_of: [ String, AwsNetworkAcl, AWS::EC2::NetworkACL ]
```

Comment:
On further investigation I believe the SDK is going to limit us here. There's no ability to simply disassociate a network acl. I even tried setting `network_acl = nil` instead of an id but it gave me an error. If anyone has ideas on how we might do that, I'm open to suggestions.